src/dir_entry.rs

```diff
@@ -83,11 +83,22 @@ impl DirEntry {
     }
 
     /// Returns true if the entry has the depth greater or equal compared to `depth`
-    pub fn deeper(&self, depth: usize) -> bool {
+    pub fn deeper(&self, depth: usize, root_paths: &[PathBuf]) -> bool {
         match &self.inner {
             DirEntryInner::Normal(e) => e.depth() >= depth,
-            // broken symlink passes match if it is just longer thatn the `depth`
-            DirEntryInner::BrokenSymlink(p) => p.components().count() - 1 >= depth,
+            DirEntryInner::BrokenSymlink(path) => {
+                for root_path in root_paths {
```

Comment:
I think this is wrong for things like `fd . foo foo/bar`

Reply:
What could go wrong, roots nesting in each other? Searching for a dir (`fd -L fourth foo foo/bar` ) and a broken link (`fd -L broken foo foo/bar`) works the same way for layout like
```foo/
foo/bar/
foo/bar/baz/
foo/bar/baz/fourth/
foo/bar/baz/broken
```

Reply:
Thanks for the tip, I forgot to specify depth.

That's right, current implementation will output more than one entry for this configuration. At least, it is not empty output.

Reply:
I have rethought the PR a little, according to the latest `ignore` and `fd` changes. I think that correct depth tracking with `WalkParallel` invoked with multiple paths is impossible. If only one thread in the walker exists, one can track depth with a root `DirEntry`. The best way to solve this is to have a "root" entry passed by the 'ignore' crate to each `FnVisitor` call alongside a `DirEntry`. It's unlikely to change `ignore` crate in this way since it's a breaking API change.

So far, PR solves the symlink problem with the best effort, and I cannot come up with a better idea.